frontend-web/webclient/app/Files/Uploader.tsx

```diff
@@ -80,6 +80,36 @@ async function processUpload(upload: Upload) {
     upload.initialProgress = reader.offset;
     upload.fileSizeInBytes = reader.fileSize();
 
+    upload.resume = createResumeable(reader, upload, strategy, fullFilePath);
+    await upload.resume();
+}
+
+function createResumeable(
+    reader: ChunkedFileReader,
+    upload: Upload,
+    strategy: FileApi.FilesCreateUploadResponseItem,
+    fullFilePath: string
+) {
+    return async () => {
+        while (!reader.isEof() && !upload.terminationRequested) {
+            await sendChunk(await reader.readChunk(maxChunkSize));
+
+            const expiration = new Date().getTime() + FOURTY_EIGHT_HOURS_IN_MILLIS;
+            localStorage.setItem(
+                createLocalStorageUploadKey(fullFilePath),
+                JSON.stringify({chunk: reader.offset, size: upload.fileSizeInBytes, response: strategy!, expiration} as LocalStorageFileUploadInfo)
```

Comment:
It might be better if we rename `chunk` to `offset` since `chunk` might indicate that we are counting the number of chunks we have sent as opposed to the number of bytes (which I am pretty sure is what `reader.offset` reports).

I also think it would make more sense to rename `response` to `strategy`.